### ISSUE TITLE
Unsupported loss function in seq2seq model.

### ISSUE THREAD

**Issue by radi-cho** [tensorflow/tensorflow#44342]
I am exploring the following (official Google DataLab) tensorflow example: https://github.com/googledatalab/notebooks/blob/master/samples/TensorFlow/LSTM%20Punctuation%20Model%20With%20TensorFlow.ipynb which apparently is written in tf v1, so I upgraded with the v2 upgrade script and there were three main inconsistencies: 

```
ERROR: Using member tf.contrib.rnn.DropoutWrapper in deprecated module tf.contrib. tf.contrib.rnn.DropoutWrapper cannot be converted automatically. tf.contrib will not be distributed with TensorFlow 2.0, please consider an alternative in non-contrib TensorFlow, a community-maintained repository such as tensorflow/addons, or fork the required code.
ERROR: Using member tf.contrib.legacy_seq2seq.sequence_loss_by_example in deprecated module tf.contrib. tf.contrib.legacy_seq2seq.sequence_loss_by_example cannot be converted automatically. tf.contrib will not be distributed with TensorFlow 2.0, please consider an alternative in non-contrib TensorFlow, a community-maintained repository such as tensorflow/addons, or fork the required code.
ERROR: Using member tf.contrib.framework.get_or_create_global_step in deprecated module tf.contrib. tf.contrib.framework.get_or_create_global_step cannot be converted automatically. tf.contrib will not be distributed with TensorFlow 2.0, please consider an alternative in non-contrib TensorFlow, a community-maintained repository such as tensorflow/addons, or fork the required code.
```

So for compatibility I manually replaced `framework.get_or_create_global_step` with `tf.compat.v1.train.get_or_create_global_step`, and also `rnn.DropoutWrapper` with `tf.compat.v1.nn.rnn_cell.DropoutWrapper`.

But I was unable to find a solution on how to handle the `tf.contrib.legacy_seq2seq.sequence_loss_by_example` method, since I cannot find a backwards compatible alternative. I tried installing Tensroflow Addons and use [its seq2seq loss function](https://www.tensorflow.org/addons/api_docs/python/tfa/seq2seq/sequence_loss), but wasn't able to figure out how to adapt it to work with the rest of the code.

Stumbled across some errors like `Consider casting elements to a supported type.` or `Logits must be a [batch_size x sequence_length x logits] tensor`, because probably i am not implementing something correctly.

**My question:** How to implement supported tensorflow v2 alternative of this loss function, so it acts similarly to the code below?

```python
    output = tf.reshape(tf.concat(axis=1, values=outputs), [-1, size])
    softmax_w = tf.compat.v1.get_variable("softmax_w", [size, len(TARGETS)], dtype=tf.float32)
    softmax_b = tf.compat.v1.get_variable("softmax_b", [len(TARGETS)], dtype=tf.float32)
    logits = tf.matmul(output, softmax_w) + softmax_b
    self._predictions = tf.argmax(input=logits, axis=1)    
    self._targets = tf.reshape(input_.targets, [-1])
    loss = tfa.seq2seq.sequence_loss(
        [logits],
        [tf.reshape(input_.targets, [-1])],
        [tf.ones([batch_size * num_steps], dtype=tf.float32)])
    self._cost = cost = tf.reduce_sum(input_tensor=loss) / batch_size
    self._final_state = state
```

*Full code [here](https://github.com/googledatalab/notebooks/blob/master/samples/TensorFlow/LSTM%20Punctuation%20Model%20With%20TensorFlow.ipynb).*

**Comment by jvishnuvardhan** [comment]
@radi-cho I noticed that you have created same issue in Google DataLab repo. As this is more related to `Google DataLab`, I am closing this issue from tensorflow repo. Thanks!

Tensorflow github is mainly for bug and performance. You could post this issue in Stackoverflow where there is a large community to support this kind of questions. Thanks!

**Comment by radi-cho** [comment]
@jvishnuvardhan Thank you as well :)
I posted it here because as I see, the datalab repo is outdated and not supported since last year.
I also tried getting help on stackoverflow: https://stackoverflow.com/questions/64522160/tensorflow-v2-alternative-of-sequence-loss-by-example, but did not get any replies. I hoped I could find some help here :)

**Comment by skwolvie** [comment]
any solution?
